feature-libs/product/configurators/common/src/components/service/config-utils.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import {
+  Configurator,
+  ConfiguratorGroupsService,
+  GenericConfigurator,
+} from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { map, take } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ConfigUtilsService {
+  constructor(private configuratorGroupsService: ConfiguratorGroupsService) {}
+
+  /**
+   * Does the configuration belong to a cart entry, or has the group been visited already?
+   * In both cases we need to render indications for mandatory attributes
+   * @param owner
+   * @param groupId
+   */
+  isCartEntryOrGroupVisited(
+    owner: GenericConfigurator.Owner,
+    groupId: string
+  ): Observable<boolean> {
+    return this.configuratorGroupsService.isGroupVisited(owner, groupId).pipe(
+      take(1),
```

Comment:
Just curious, why `take(1)` here?
I'm asking, because the callers of `isCartEntryOrGroupVisited` will not be able to receive a new value emmited from `this.configuratorGroupsService.isGroupVisited`, as `take(1)` will unsubscribe from it

Reply:
Good question, I took a look into it: It is indeed ok here, because during the lifetime of components that use this API they only need one emission, as this state cannot change. Only when you navigate to another group (means lifetime of attribute components is over), the value will change. I will mention this in the JS Doc